src/API.php

```diff
@@ -44,42 +44,76 @@ public function login($username, $password, $uuid = null)
         return $login['success'];
     }
 
-  /**
-  * getFiles returns the files for the user account
-  *
-  * @return array files
-  */
+    /**
+    * getFiles returns the files for the user account
+    *
+    * @return array files
+    */
     public function getFiles()
     {
-        return $this->request('/bot/files');
+        $files = $this->request('/bot/files');
+        $_out = array_filter($files, function ($file) {
+            return $file["parent"] === "";
+        });
+        $out = [];
+        foreach ($_out as $file) {
+            if ($file["type"] === "folder") {
+                array_push($out, new Folder($this, $file));
+            } else {
+                array_push($out, new File($this, $file));
+            }
+        }
+   
+        $todo = array_filter($files, function ($file) {
```

Comment:
Instead of calling `array_filter` twice and `foreach` twice you could also do the same thing with two `foreach` loops.
In the first one you can check if `$file["parent"] === ""`, `array_push` it as Folder or File to `$parents`, if not you can add it to `$items` and loop over them in a second `foreach` like you did below.